### DISCUSSION TITLE
new packages not compiling

### DISCUSSION THREAD

**Discussion by Morgosh** [zkSync-Community-Hub/zksync-developers#238] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.13

### zksolc/zkvyper version?

latest

### Explain the issue

```bash
Hello guys, last couple days there were many package updates on zksync, though while I understand you are still working on it, but compiling contracts just doesn't work whatever I try, for me and many other developers, I would appreciate if some solution is presented in dev-general in discord on which packages to use for it to work.

    "@matterlabs/hardhat-zksync-deploy": "^1.1.0",
    "@matterlabs/hardhat-zksync-solc": "^1.0.3",
    "@matterlabs/hardhat-zksync-verify": "^1.2.0",
    "ethers": "^6.9.0",
    "zksync-ethers": "^6.0.0"

TypeError: Class extends value undefined is not a constructor or null
```

**Comment by SebaBau** [comment]
Yeah, like @Morgosh wrote I have a lot of problems right now for example with upgradeable contracts (looks like problem with zksync-ethers).

**Comment by MexicanAce** [comment]
Hey @Morgosh , thank you for the report! We're investigating and will post updates here

**Reply by nooma-42** [reply]
Hey @MexicanAce  any update on this? I still encounter same issue 

update:
In my case, after installing `"@nomicfoundation/hardhat-verify": "^2.0.3",` the error, `TypeError: Class extends value undefined is not a constructor or null` magically solve, but the cause is still unclear to me. maybe zksync team could investigate further.

**Comment by mpopovac-txfusion** [comment]
Hey @Morgosh, I tried to replicate the error, and I didn't succeed. From the dependencies mentioned in the issue, the only one missing is hardhat. When I added that, it successfully compiled the contracts and deployed them. The only thing that crosses my mind here is that there might be an issue with dependencies. The error stated here, `TypeError: Class extends value undefined is not a constructor or null`, arises when you have installed zksync-ethers but haven't installed the ethers dependency. 
Could you please provide the repository link? Or at least the hardhat.config file and package.json?

**Reply by Morgosh** [reply, marked as answer]
I excluded the contract/hardhat part from my project and was able to get it to work, was using yarn monorepo, so I can't really figure out which package was conflicting. But can confirm that I got it working by having latest packages and making sure there are no conflicting ones:

    "@matterlabs/hardhat-zksync-deploy": "^1.1.0",
    "@matterlabs/hardhat-zksync-solc": "^1.0.3",
    "@matterlabs/hardhat-zksync-verify": "^1.2.0",
    "@matterlabs/zksync-contracts": "^0.6.1",
    "ethers": "^6.9.0",
    "hardhat": "^2.19.2",
    "zksync-ethers": "^6.0.0",

by running:

yarn add -D hardhat @matterlabs/hardhat-zksync-deploy @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-verify @matterlabs/zksync-contracts ethers zksync-ethers

Also make sure to uninstall zksync-web3

**Comment by SebaBau** [comment]
But still there are some errors for example with tests. Errors occurs when "zksync-ethers" is installed. Also still is the problem with upgradeable contracts...

**Reply by mpopovac-txfusion** [reply]
Hey, this is probably connected to the way you are using it because there are some changes in the interface between ethers v5 and ethers v6. I would strongly recommend going through the migration guide (https://docs.ethers.org/v6/migrating/, https://era.zksync.io/docs/api/js/zksync2-js/migration.html).

The PR for the docs with updated examples is merged, but the documentation is yet to be released. Nonetheless, here is the PR link so you can check it out and see if this can resolve your issues: https://github.com/matter-labs/zksync-web-era-docs/pull/823/files

**Reply by Morgosh** [reply]
zkSync documentation is yet to be updated, I had the same issue with tests, where I got 
Cannot read properties of undefined (reading 'url')

But found a temporal solution by adding the following to config:
    "http://localhost:8545": {
      url: "http://localhost:3050",
      ethNetwork: "http://localhost:8545",
      zksync: true,
    }
   
   naming is important since the new deployer seems to always read from this config name "http://localhost:8545"

**Reply by SebaBau** [reply]
Yeap, that solution works :)

**Reply by SebaBau** [reply]
Thank you for the answer, @Morgosh helped me a lot, looks like this "workaround" with Hardhat config is necesarry.

**Comment by ljada37** [comment]
It seems like there might be compatibility issues with the latest updates to the zksync packages and your current development environment. One potential solution could be to roll back to earlier versions of the zksync packages until the compatibility issues are resolved. You can try downgrading the zksync packages to versions that were working for you previously and monitor the project's GitHub repository or Discord channel for updates on when the issues are fixed.

**Comment by heuresfanegarnold** [comment]
It seems like there might be compatibility issues or changes in the dependencies that are causing the compilation problem. 

Here are some steps you can take to address this issue:
1. Check the documentation or release notes of the zksync packages to see if there are any recent updates or changes that might affect compatibility.
2. Make sure you are using compatible versions of all dependencies, including hardhat, zksync packages, ethers, and zksync-ethers.
3. Try downgrading or upgrading specific packages to versions that are known to work together.
4. If possible, consult the community forums or channels like dev-general on Discord to see if other developers are experiencing similar issues and if any workarounds or solutions have been identified.
5. Consider reaching out to the maintainers of the zksync packages for assistance or clarification on the issue.

By carefully reviewing the dependencies and seeking support from the community or maintainers, you should be able to resolve the compilation problem and continue with your development work.

**Comment by tumb2ibonsa** [comment]
The error "TypeError: Class extends value undefined is not a constructor or null" typically indicates a problem with the dependencies or the way classes are extended in your code. It's possible that there is a compatibility issue between the versions of the packages you are using. Please ensure that all dependencies are updated and compatible with each other. Additionally, consider checking the documentation or seeking support from the package maintainers for guidance on resolving this issue.

**Comment by dyksterhouset** [comment]
Compiler: zksolc
Solidity Version: 0.8.13
zksolc Version: latest
Issue: Error message "TypeError: Class extends value undefined is not a constructor or null" encountered while compiling contracts after recent package updates on zksync.

**Comment by wadejackie** [comment]
The user is encountering an issue while attempting to compile contracts using zksync packages. They have provided a list of the packages they are using and the error message they are receiving, which indicates a TypeError related to a class extension. The user has not provided a repository link. Further investigation may be needed to determine the cause of the error and potential solutions.
